src/tests/functional/plugin/cpu/single_layer_tests/interaction.cpp

```diff
@@ -0,0 +1,125 @@
+        std::vector<int32_t> reshape_value = {3, 27, 4};
+        auto reshape_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{3}, reshape_value);
+        auto reshape = std::make_shared<opset1::Reshape>(concat1, reshape_shape, true);
+        std::vector<int32_t> transpose1_value = {0, 2, 1};
+        auto transpose1_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{3}, transpose1_value);
+        auto transpose1 = std::make_shared<opset1::Transpose>(reshape, transpose1_shape);
+        auto matmul = std::make_shared<opset1::MatMul>(reshape, transpose1);
+        std::vector<int32_t> transpose2_value = {1, 2, 0};
+        auto transpose2_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{3}, transpose2_value);
+        auto transpose2 = std::make_shared<opset1::Transpose>(matmul, transpose2_shape);
+        std::vector<int32_t> reshape2_value = {729, -1};
+        auto reshape2_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{2}, reshape2_value);
+        auto reshape2 = std::make_shared<opset1::Reshape>(transpose2, reshape2_shape, true);
+
+        std::vector<int32_t> gather_indices_value;
+        for (int i = 1; i < 27; i++) {
+            for (int j = 0; j < i; j ++) {
+                gather_indices_value.push_back(i * 27 + j);
+            }
+        }
+        auto gather_indices =  std::make_shared<opset1::Constant>(element::i32, Shape{351}, gather_indices_value);
+        auto gather_axis =  std::make_shared<opset1::Constant>(element::i32, Shape{}, 0);
+        auto gather = std::make_shared<opset8::Gather>(reshape2, gather_indices, gather_axis);
+        std::vector<int32_t> reshape3_value = {-1, 3};
+        auto reshape3_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{2}, reshape3_value);
+        auto reshape3 = std::make_shared<opset1::Reshape>(gather, reshape3_shape, true);
+
+        std::vector<int32_t> transpose3_value = {1, 0};
+        auto transpose3_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{2}, transpose3_value);
+        auto transpose3 = std::make_shared<opset1::Transpose>(reshape3, transpose3_shape);
+
+        std::vector<int32_t> reshape4_value = {3, 351};
+        auto reshape4_shape =  std::make_shared<opset1::Constant>(element::i32, Shape{2}, reshape4_value);
+        auto reshape4 = std::make_shared<opset1::Reshape>(transpose3, reshape4_shape, true);
+        auto concat2 = std::make_shared<opset1::Concat>(NodeVector{dense_feature, reshape4}, 1);
+        auto relu = std::make_shared<opset1::Relu>(concat2);
+        function = std::make_shared<ov::Model>(relu, ov::ParameterVector{dense_feature}, "interaction");
+    }
+};
+
+TEST_P(IntertactionLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    run();
+    CheckPluginRelatedResults(compiledModel, "interaction");
+}
+
+namespace {
+const std::vector<ElementType> inPrecisions = {
+        ElementType::f32,
```

Comment:
Please add at least i32 case.

Reply:
Hi the inputs of interaction are embedding features, it could only be float precision. I think I32 is not supported.

Reply:
You need to fix the fn *initSupportedPrimitiveDescriptors* first, then this test will check operation with converted data.

Reply:
Sure, I get your point, this case is used for precision conversion, I'll add it.

Reply:
Add precision conversion in c67fb3cc606836770a72b723ec7843f237c2864a